src/app/blocks/http-block/http-block.component.ts

```diff
@@ -130,126 +133,220 @@ export class HttpBlockComponent implements OnInit, OnChanges {
+            return EMPTY;
+          }
+        }),
+        takeWhile(response => response.status === 200), // Stop when there's an error or nextPageUrl is null
+        catchError(error => {
+          this.hasError = true;
+          this.errorMessage = error.message;
+          this.errorData = error;
+          return of([]);
+        }),
+        reduce((accumlated_results: any[], response: HttpResponse<any>) => accumlated_results.concat(response.body || []), [])
+      )
+      .subscribe(results => {
+        this.isLoading = false;
+        this.hasError = false;
+        this.outputResult(results);
+      });
+  }
 
+  /**
+   * Extracts the next page URL from the link header in the HttpResponse.
+   * @param {string} linkHeader - The link header string from HttpResponse.
+   * @return {string|null} - The next page URL if it exists, or null otherwise.
+   */
+  extractNextPageUrl(linkHeader) {
+    /**
+     * The `link` header should follow the format outlined by RFC 5988 Web Linking. 
+     * It should contain a list of link relations and their respective URLs, separated 
+     * by commas. Each link consists of the URL enclosed in angle brackets `< >` 
+     * followed by a semicolon and the relation type specified as `rel="relation-type"`.
+     * 
+     * Example of a link header:
+     * `<linkHeader> = '<https://example.com/data?page=2>; rel="next", 
+     *                 <https://example.com/data?page=5>; rel="last"'`
+     * 
+     * In this example, there is a "next" relation pointing to the second page of results,
+     * and a "last" relation pointing to the fifth (and final) page of results.
+     * 
+     * The `extractNextPageUrl` method will specifically look for a link with the `rel="next"` 
+     * attribute and return its URL. In case there's no "next" relation, the method will 
+     * return null, which is considered the stop condition for pagination. 
+     */
+    if (!linkHeader) {
+      return null;
     }
 
+
+    // We use a regex to extract the URL and relation type from each link,
+    // and return an object with the URL and relation type.
+    const links = linkHeader.split(',').map(link => {
+      // The regex will match the first pair of angle brackets enclosing the URL,
+      // and the first pair of double quotes enclosing the relation type.
+      // Example: `<https://example.com/data?page=2>; rel="next"`
+      // The regex will match `<https://example.com/data?page=2>` and `next`
+      // and return an object with the URL and relation type.
+
+      const matches = /<(.*)>; rel="(.*)"/.exec(link.trim());
+      if (matches && matches.length === 3) {
+        return { url: matches[1], rel: matches[2] };
+      }
+    });
+
+    // Get the first link that matches the condition:
+    const nextPageLink = links.find(link => link && link.rel === 'next');
+    // We return the URL of the next page if it exists, or null otherwise.
+    return nextPageLink ? nextPageLink.url : null;
```

Comment:
Cool, optional chaining is nice. As a general best practice, I think it's best to avoid undefined behaviour. So I went with `return nextPageLink?.url || null;` which seems most conventional and most readable to me.